File: 167. Two Sum II - Input Array Is Sorted/twoSum.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (numbers, target) {
  let index1 = 1;
  let index2 = numbers.length;
  while (index1 !== index2) {
    if (numbers[index1 - 1] + numbers[index2 - 1] < target) {
      index1++;
    } else if (numbers[index1 - 1] + numbers[index2 - 1] > target) {
      index2--;
    } else {
      return [index1, index2];
    }
  }
};

File: 1. Two Sum/v1/twoSum.go
package v1

func twoSum(nums []int, target int) []int {
	if len(nums) == 2 {
		return []int{0, 1}
	}
	for i := 0; i < len(nums)-1; i++ {

		for j := i + 1; j < len(nums); j++ {

			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return nil
}
